### ISSUE TITLE
Risk of NaN loss in Pi0's sampling_beta implementation, especially using bfloat16

### ISSUE THREAD

**Issue by YuhengZhi** [huggingface/lerobot#1096]
### System Info

```Shell
Not related.
```

### Information

- [x] One of the scripts in the examples/ folder of LeRobot
- [ ] My own task or dataset (give details below)

### Reproduction

In the current implementation of `sample_beta` in `modeling_pi0.py` ([link](https://github.com/huggingface/lerobot/blob/a445d9c9da6bea99a8972daa4fe1fdd053d711d2/lerobot/common/policies/pi0/modeling_pi0.py#L92-L95)):
```python
def sample_beta(alpha, beta, bsize, device):
    gamma1 = torch.empty((bsize,), device=device).uniform_(0, 1).pow(1 / alpha)
    gamma2 = torch.empty((bsize,), device=device).uniform_(0, 1).pow(1 / beta)
    return gamma1 / (gamma1 + gamma2)
```
When `gamma1` and `gamma2` are both sampled to be 0, the returned value will be `NaN`, causing loss to be also `NaN`. This happens very rarely in FP32 but when I attempted to train the pi0 model fully in `bf16`, it becomes much easier for `gamma1` and `gamma2` to be 0. 

Consequently, **my loss becomes `NaN`** after roughly the first 24k samples.

This is a bug that may arise when a user tries either:
a) changing Pi0's precision fully to `bfloat16` and training for ~24k samples, or
b) training pi0 in the default mixed precision for a very large number of steps.

A similar [issue](https://github.com/stan-dev/math/issues/419) was brought up in the C++ Stan Math Library. They modified implementation of beta distribution to _happen in the log space_, which is numerically stable, as similarly done by the official openpi code ([openpi's code](https://github.com/Physical-Intelligence/openpi/blob/36dc3c037eb8a3921be9ecb94369d60cbf56f58f/src/openpi/models/pi0.py#L249), [`jax.random.beta` implementation](https://github.com/jax-ml/jax/blob/ddfcf84a375677e4dccfa92ddc2b4a7eb7234560/jax/_src/random.py#L1021-L1038)). 

### Expected behavior

`sample_beta` should never return `NaN`.

**Comment by lin-whale** [comment]
follow

**Comment by pkooij** [comment]
Hi we update our pi0 implementation and released pi05 support in this PR: https://github.com/huggingface/lerobot/pull/1910 🚀 . If you still have this issue please reopen this thread, but I am closing it for now.
